Mage.Sets/src/mage/cards/p/Plaguecrafter.java

```diff
@@ -0,0 +1,94 @@
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.common.discard.DiscardTargetEffect;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.Outcome;
+import mage.constants.SubType;
+import mage.filter.FilterPermanent;
+import mage.filter.StaticFilters;
+import mage.filter.common.FilterCreatureOrPlaneswalkerPermanent;
+import mage.filter.predicate.permanent.ControllerIdPredicate;
+import mage.game.Game;
+import mage.target.targetpointer.FixedTarget;
+
+/**
+ *
+ * @author themogwi
+ */
+public final class Plaguecrafter extends CardImpl {
+
+    public Plaguecrafter(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId,setInfo,new CardType[]{CardType.CREATURE},"{2}{B}");
+        this.subtype.add(SubType.HUMAN);
+        this.subtype.add(SubType.SHAMAN);
+
+        this.power = new MageInt(3);
+        this.toughness = new MageInt(2);
+
+        // When Plaguecrafter enters the battlefield.
+        //   Each player sacrifices a creature or planeswalker.
+        //   Each player who can't discards a card.
+        this.addAbility(new EntersBattlefieldTriggeredAbility(new PlaguecrafterEffect()));
+    }
+
+    public Plaguecrafter(final Plaguecrafter card) {
+        super(card);
+    }
+
+    @Override
+    public Plaguecrafter copy() {
+        return new Plaguecrafter(this);
+    }
+}
+
+
+class PlaguecrafterEffect extends OneShotEffect {
+
+    public PlaguecrafterEffect() {
+        super(Outcome.Benefit);
+        this.staticText = "Each player sacrifices a creature or planeswalker. "
+                + "Each player who can't discards a card.";
+    }
+
+    public PlaguecrafterEffect(final PlaguecrafterEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public PlaguecrafterEffect copy() {
+        return new PlaguecrafterEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        game.getPlayers().forEach((playerId, player) -> {
+            if (!(player == null)) {
+                FilterPermanent filter = new FilterCreatureOrPlaneswalkerPermanent();
+                filter.add(new ControllerIdPredicate(playerId));
+                if (game.getBattlefield().getActivePermanents(
+                        filter, source.getControllerId(), game
+                ).isEmpty()) {
+                    Effect discardEffect = new DiscardTargetEffect(1);
+                    discardEffect.setTargetPointer(new FixedTarget(playerId, game));
+                    discardEffect.apply(game, source);
+                } else {
+                    Effect effect = new SacrificeEffect(
+                            StaticFilters.FILTER_PERMANENT_CREATURE_OR_PLANESWALKER_A, 1, null
+                    );
+                    effect.setTargetPointer(new FixedTarget(playerId, game));
+                    effect.apply(game, source);
+                }
+            }
+        });
```

Comment:
From the [release notes](https://magic.wizards.com/en/articles/archive/feature/guilds-ravnica-release-notes-2018-09-20):

> As Plaguecrafter's ability resolves, first the player whose turn it is chooses a creature or planeswalker they control, then each other player in turn order does the same, knowing the choices made before them. Then all the chosen permanents are sacrificed at the same time. Next, each player in the same order who couldn't sacrifice a permanent chooses a card in hand without revealing it, then the chosen cards are discarded at the same time.

You need to change the order in which the events occur.